### PR TITLE
Bring commonly used builtin options under lib directory

### PR DESCRIPTION
**Pull request by kamleshbhalui**

### CODE REVIEW

File: lib/options.cpp [hunk 1]
```diff
@@ -47,6 +47,18 @@ std::vector<const char*>* Util::Options::process(int argc,
     if (argc == 0 || argv == nullptr)
         throw std::logic_error("No arguments to process");
     binaryName = argv[0];
+    // collect command line args
+    for (int i = 0; i < argc; i++) {
+        compileCommand += argv[i];
+        compileCommand += " ";
+    }
+    compileCommand = compileCommand.replace("(from pragmas)", "");
```
Comment (mbudiu-vmw) [inline lib/options.cpp:55]: Where is this coming from? Why does it apply to all backends?

Comment (kamleshbhalui) [reply lib/options.cpp:55]: it is required here (dpdk backend) and other backend adopting from base code would need
and this information is available(first hand here).

Comment (mbudiu-vmw) [reply lib/options.cpp:55]: Where is the string "from pragmas" coming from?

Comment (kamleshbhalui) [reply lib/options.cpp:55]: frontends/common/applyOptionsPragmas.cpp:30

Comment (mbudiu-vmw) [reply lib/options.cpp:55]: Wouldn't it be cleaner to just skip "(from pragmas)" only if it is argv[0]?

Comment (kamleshbhalui) [reply lib/options.cpp]: yes skipped it.

**Review by mbudiu-vmw — APPROVED** [review]

**Pull request merged**